client/src/partials/patients/documents/modals/documents.modal.js

```diff
@@ -0,0 +1,53 @@
+angular.module('bhima.controllers')
+.controller('PatientDocumentsModalController', PatientDocumentsModalController);
+
+PatientDocumentsModalController.$inject = [
+  '$uibModalInstance', 'Upload', 'patientUuid', 'PatientService', 'NotifyService'
+];
+
+function PatientDocumentsModalController(Instance, Upload, patientUuid, PatientService, Notify) {
+  var vm = this;
+
+  /** expose to the view */
+  vm.submit  = submit;
+  vm.close   = close;
+
+  /** init require data  */
+  PatientService.read(patientUuid)
+  .then(function (patient) {
+    vm.patient = patient;
+  });
+
+  /** close the modal */
+  function close() {
+    Instance.dismiss('cancel');
+  }
+
+  /** submit data on server */
+  function submit(form) {
+    if (vm.file) {
+      uploadFile(vm.file);
+    }
+  }
+
+  // upload on file select or drop
+  function uploadFile (file) {
+    var title = vm.title || file.name;
+    Upload.rename(file, title);
+    Upload.upload({
+        url: '/patients/' + patientUuid + '/documents',
+        data: { documents: file }
+    }).then(function (resp) {
+      // success
+      vm.uploaded = true;
+      Notify.success('PATIENT_DOCUMENT.UPLOAD_SUCCESS');
+      Instance.close();
+    }, function (resp) {
+      // error
```

Comment:
For the error case, you should try to use `Notify.handleError`.  Does it work in this case?